pkg-manager/core/test/install/peerDependencies.ts

```diff
@@ -1412,7 +1412,7 @@ test('resolve peer dependency from aliased direct dependency', async () => {
   await addDependenciesToPackage(manifest, ['@pnpm.e2e/abc@1.0.0'], opts)
 
   const lockfile = await readYamlFile<any>(path.resolve(WANTED_LOCKFILE)) // eslint-disable-line
-  expect(lockfile.packages).toHaveProperty(['/@pnpm.e2e/abc@1.0.0(@pnpm.e2e/peer-a@1.0.0)'])
+  expect(lockfile.packages['/@pnpm.e2e/abc@1.0.0(@pnpm.e2e/peer-a@1.0.0)']).toBeTruthy()
```

Comment:
why was this changed?

Reply:
We shouldn't use `toHaveProperty` if there's a dot in the property, it'll be treated as nested property such as `user.name`.

Reply:
I don't think so because it works

Reply:
For `user.name` you'd do `.toHaveProperty(['user', 'name'])`

Reply:
It doesn't work when I run tests on my PC:

```text
  ● when there are several aliased dependencies of the same package, pick the one with the highest version to resolve peers                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                                                                                                 
    expect(received).toHaveProperty(path)

    Expected path: "/@pnpm.e2e/abc@1.0.0(@pnpm.e2e/peer-c@2.0.0)"
    Received path: []

    Received value: {"/@pnpm.e2e/abc@1.0.0(@pnpm.e2e/peer-c@2.0.0)": {"dependencies": {"@pnpm.e2e/dep-of-pkg-with-1-dep": "100.0.0", "@pnpm.e2e/peer-c": "2.0.0"}, "dev": false, "peerDependencies": {"@pnpm.e2e/peer-a": "^1.0.0", "@pnpm.e2e/peer-b": "^1.0.0", "@pnpm.e2e/peer-c": "^1.0.0"}, "resolution": {"integrity": "sha512-/BCMmCRQ58DLWtF5iU97Lbf0s+llpWHvvZvQfFnCYFBpa1pfYnlJW1k6+Z4UzcUdYrUvyDmYPwxGVvPStAUw1w=="}}, "/@pnpm.e2e/dep-of-pkg-with-1-dep@100.0.0": {"dev": false, "resolution": {"integrity": "sha512-KUmwlEiE3pzpmPCfrLkoOO7fmL4Tft+dcm6YQlUfOUJvHO73xlcKVoA/xOHP/ayUO8GiPlIqWRbKHrsxvXGE8g=="}}, "/@pnpm.e2e/peer-c@1.0.0": {"dev": false, "resolution": {"integrity": "sha512-SQ3TfmnPopiVIqzC/PVV8WlfJjVWFdszQhfZG6e9stDFD9i9dZ5VQImKYhvJsME4nb45rHGsc/4CmroEjO7aCA=="}}, "/@pnpm.e2e/peer-c@1.0.1": {"dev": false, "resolution": {"integrity": "sha512-6ag+AQOLJgsioiG8aInnOJGautKz7lNf3zKQ8FJHOb4b3yIclmUofTaDwth9ydp9W56qzwbHWXjkdrqfW5JViQ=="}}, "/@pnpm.e2e/peer-c@2.0.0": {"dev": false, "resolution": {"integrity": "sha512-s0TPtcya7A2r7vSzmq1dNAdxUNY0TQXeyARH8YMe/d7wb3/I4gryxBP3VVGPuKf6LUO+AALCBr0Jha8wwG7isg=="}}}

      1424 |
      1425 |   const lockfile = await readYamlFile<any>(path.resolve(WANTED_LOCKFILE)) // eslint-disable-line
    > 1426 |   expect(lockfile.packages).toHaveProperty('/@pnpm.e2e/abc@1.0.0(@pnpm.e2e/peer-c@2.0.0)')
           |                             ^
      1427 | })
      1428 |

      at Object.<anonymous> (test/install/peerDependencies.ts:1426:29)
```

The [Jest docs](https://jestjs.io/docs/expect#tohavepropertykeypath-value) says, "For checking deeply nested properties in an object you may use [dot notation](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Property_accessors) or an array containing the keyPath for deep references".